api/client/copy.go

```diff
@@ -0,0 +1,218 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/pkg/archive"
+)
+
+type copyDirection int
+
+const (
+	fromContainer copyDirection = (1 << iota)
+	toContainer
+	acrossContainers = fromContainer | toContainer
+)
+
+// CmdCp handles copying files or directories to/from/across containers.
+func (cli *DockerCli) CmdCp(args ...string) error {
+	cmd := cli.Subcmd("cp", "[SRC_CONTAINER:]SRC_PATH [DST_CONTAINER:]DST_PATH", "Copy files between containers or a local path", true)
```

Comment:
Note that this will fail on windows due to the colon. Sadly we can't even assume that a container name longer than a single char *isn't* a drive on windows because technically you can identify a container by just a single char.